prompt2model/dataset_retriever/description_dataset_retriever.py

```diff
@@ -341,16 +420,38 @@ def retrieve_top_datasets(
     def retrieve_dataset_dict(
         self,
         prompt_spec: PromptSpec,
+        data_transform: bool = False,
```

Comment:
Can we rename this to focus on a verb rather than a noun?

e.g. `auto_transform_data`